### PR TITLE
Cluwnebans

### PR DESCRIPTION
**Pull request by ghost**
how it works:
- cluwnebanned players can only latejoin as a cluwne
- cluwnebanned players cannot use or read (L)OOC
- cluwnebanned players are kicked upon death, and cannot rejoin the game until the next round
- cluwnebanned players can not have their minds swapped or be transformed in any way

(sneaky cluwnebanned players can click observe while the server is initializing, but because this sets their time of death it also immediately kicks them from the server and prevents rejoining)

this is necessarily(?) hacky but seems to work fine

### CODE REVIEW

**Comment by Dacendeth** [comment]
What if cluwne goes into a mindswap machine or a wizard shoots them with a staff of change?

**Comment by ghost** [comment]
Those (and the mindswap spell) are both very easy to add checks to, I'll get around to them in a bit.

File: code/modules/mob/new_player/new_player.dm [hunk 1]
```diff
@@ -157,6 +161,15 @@
 				return 0
 
 		LateChoices()
+	if(href_list["cluwnebanned"])
+		if(!ticker || ticker.current_state <= GAME_STATE_PREGAME)
```
Comment (Kurfursten) [inline code/modules/mob/new_player/new_player.dm:168]: due to lack of sanity, this allows you to use href exploits to join as a cluwne even if not banned

**Comment by SageOfLight** [comment]
I can honk to this.

**Comment by KingofHONK** [comment]
yesss...hahahaahahaa....YESSSSS

**Comment by ghost** [comment]
Barred cluwnebanned players from mindswap, staff of change, and mind machine. If there's any IC methods of transforming/mind transfer I'm forgetting please let me know.

**Comment by GreatBigFailure** [comment]
Can you force feed cluwnes? Test wendigo meat, royal gingjelly, and slime toxin on them to make sure.

**Comment by ghost** [comment]
Tested, they can't eat or be fed and they don't metabolize any of the reagents that cause transformations.

**Comment by Exxion** [comment]
If you want to be super extra thorough you can just check in `/mob/Login()` if the client is cluwnebanned and delete the client unless the mob is a cluwne (or rather has a var set that lets cluwnebanned players enter it or whatever)

**Comment by Kurfursten** [comment]
Do what Exxion said and we'll merge it

**Review by Kurfursten — CHANGES_REQUESTED** [review]
What he said

**Comment by ghost** [comment]
Done

**Review by Kurfursten — APPROVED** [review]

**Comment by dunsel-hordon** [comment]
so now instead of permabanning people we cluwnebann them?

**Pull request merged**